File: src/my_py_pkg/my_py_pkg/service_server.py
import rclpy
from rclpy.node import Node
from my_robot_interfaces.srv import AddTwoInts

class AddTwoIntsServer(Node):
    def __init__(self):
        super().__init__("add_two_init_server")
        self.server_ = self.create_service(AddTwoInts, "add_two_ints", self.callback_add_two_ints)
        self.get_logger().info("Add Two Ints server started")

    def callback_add_two_ints(self, request, respone):
        respone.sum = request.a + request.b
        self.get_logger().info(f"{request.a} + {request.b} = {respone.sum}")
        return respone
    
def main(args = None):
    rclpy.init(args=args)
    rclpy.spin(AddTwoIntsServer())
    rclpy.shutdown()

if __name__ =="__main__":
    main()

File: src/my_cpp_pkg/src/service_server.cpp
#include <cinttypes>
#include <functional>
#include <memory>
#include "rclcpp/rclcpp.hpp"
#include "my_robot_interfaces/srv/add_two_ints.hpp"

class AddTwoIntsServer : public rclcpp::Node {
public:
    AddTwoIntsServer(): Node("add_two_ints_server") {
        server_ = create_service<my_robot_interfaces::srv::AddTwoInts>(
            "add_two_ints",
            std::bind(&AddTwoIntsServer::callbackAddTwoInts, this, std::placeholders::_1, std::placeholders::_2));
    }
private:
    void callbackAddTwoInts(
        const my_robot_interfaces::srv::AddTwoInts::Request::SharedPtr req,
        const my_robot_interfaces::srv::AddTwoInts::Response::SharedPtr res){
        res->sum = req->a +req->b;
        RCLCPP_INFO(get_logger(), "%" PRId64 " + %" PRId64 " =  %" PRId64, req->a, req->b, res->sum);
        }
        rclcpp::Service<my_robot_interfaces::srv::AddTwoInts>::SharedPtr server_;
        
};
int main(int argc, char **argv){
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<AddTwoIntsServer>());
    rclcpp::shutdown();
    return 0;
}